Homework Submission on Time

talha-ahsan opened the pull request:
It's not fully done. But I'm COMPLETELY STUMPED on a few parts so this is reasonably the best I'll get. NullType wiped me out.

talha-ahsan committed:
Hmoework Commit

Amostdone!

talha-ahsan committed:
Fixed two more errors!

Now on to constructors and deep calls?

talha-ahsan committed:
Fixed all but one error for type checker

SO CLOSE.

talha-ahsan committed:
REALLY BAD CODE

BUT IT WORKS. BADA BOOM BADA BANG

talha-ahsan committed:
Committing because it's almost done and I definitely can't figure out the rest of it

Yes the code is messy. Does it work? Mostly. Does TestNull give me irrational rage? Yes, yes it does. Expect me to ask for help monday.

pcantrell commented:
Hang in there a little longer! Null is just a tiny bit of work compared to check_types.


> On Feb 9, 2018, at 9:50 PM, Talha Ahsan <notifications@github.com> wrote:
> 
> It's not fully done. But I'm COMPLETELY STUMPED on a few parts so this is reasonably the best I'll get. NullType wiped me out.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/pcantrell/comp394-type-modeling/pull/11 <https://github.com/pcantrell/comp394-type-modeling/pull/11>
> Commit Summary
> 
> Hmoework Commit
> Fixed two more errors!
> Fixed all but one error for type checker
> REALLY BAD CODE
> Committing because it's almost done and I definitely can't figure out the rest of it
> File Changes
> 
> A .idea/.name <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-0> (1)
> A .idea/compiler.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-1> (22)
> A .idea/copyright/profiles_settings.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-2> (3)
> A .idea/encodings.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-3> (6)
> A .idea/misc.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-4> (6)
> A .idea/modules.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-5> (8)
> D java-type-checker/.idea/inspectionProfiles/Project_Default.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-6> (21)
> M java-type-checker/.idea/misc.xml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-7> (2)
> M java-type-checker/.idea/type-checker.iml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-8> (2)
> M java-type-checker/java_type_checker/expressions.py <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-9> (69)
> M java-type-checker/java_type_checker/types.py <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-10> (32)
> M python-attr-lookup/python-attr-lookup.iml <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-11> (12)
> M python-attr-lookup/src/plang/PythonObject.java <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-12> (23)
> M python-attr-lookup/src/plang/PythonType.java <https://github.com/pcantrell/comp394-type-modeling/pull/11/files#diff-13> (12)
> Patch Links:
> 
> https://github.com/pcantrell/comp394-type-modeling/pull/11.patch <https://github.com/pcantrell/comp394-type-modeling/pull/11.patch>
> https://github.com/pcantrell/comp394-type-modeling/pull/11.diff <https://github.com/pcantrell/comp394-type-modeling/pull/11.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/pcantrell/comp394-type-modeling/pull/11>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAXFMmzquyxPY_IJT9QijKlwVJ12_AQeks5tTRIbgaJpZM4SAyfw>.
>

talha-ahsan committed:
Hunter Herman is space Jesus

Hunter helped me bugfix. Mostly helped point out why my code didn't belong where it did, and that static_type is not the same as static_type()

AttributeError: 'function' object has no attribute 'is_subtype_of' has plagued me no longer.

stwyde commented:
Hunter Herman is space man. Ends up you need parenthesis after methods in python. something I kept forgetting which kept giving me attribute errors.
